hpat/tests/test_series.py

```diff
@@ -3627,6 +3627,132 @@ def test_impl(s, axis):
             msg = 'Method cumsum(). Unsupported parameters. Given axis: int'
             self.assertIn(msg, str(raises.exception))
 
+    def test_series_getitem_value_int(self):
+        def test_impl(S):
+            return S[2]
+
+        jit_impl = hpat.jit(test_impl)
+
+        S = pd.Series([11, 22, 33], [2, 3, 5])
+        self.assertEqual(jit_impl(S), test_impl(S))
+
+    def test_series_getitem_value_str1(self):
+        def test_impl(S):
+            return S[2]
+
+        jit_impl = hpat.jit(test_impl)
+
+        S = pd.Series([11, 22, 33], ['2', '3', '5'])
+        self.assertEqual(jit_impl(S), test_impl(S))
+
+    def test_series_getitem_value_str2(self):
+        def test_impl(S):
+            return S['2']
+
+        jit_impl = hpat.jit(test_impl)
+
+        S = pd.Series([11, 22, 33], ['2', '3', '5'])
+        self.assertEqual(jit_impl(S), test_impl(S))
+
+    def test_series_getitem_slice_int(self):
+        def test_impl(S):
+            return S[0:2]
+
+        jit_impl = hpat.jit(test_impl)
+
+        S = pd.Series([11, 22, 33], [2, 3, 5])
+        self.assertEqual(jit_impl(S), test_impl(S))
+
+    def test_series_getitem_slice_str1(self):
+        def test_impl(S):
+            return S[0:2]
+
+        jit_impl = hpat.jit(test_impl)
+
+        S = pd.Series([11, 22, 33], ['2', '3', '5'])
+        self.assertEqual(jit_impl(S), test_impl(S))
+
+    def test_series_getitem_slice_str2(self):
+        def test_impl(S):
+            return S['2':'5']
+
+        jit_impl = hpat.jit(test_impl)
+
+        S = pd.Series([11, 22, 33], ['2', '3', '5'])
+        self.assertEqual(jit_impl(S), test_impl(S))
+
+    def test_series_at_value_int(self):
+        def test_impl(S):
+            return S.at[2]
+
+        jit_impl = hpat.jit(test_impl)
+
+        S = pd.Series([11, 22, 33], [2, 3, 5])
+        self.assertEqual(jit_impl(S), test_impl(S))
+
+    def test_series_at_value_str(self):
+        def test_impl(S):
+            return S.at['2']
+
+        jit_impl = hpat.jit(test_impl)
+
+        S = pd.Series([11, 22, 33], ['2', '3', '5'])
+        self.assertEqual(jit_impl(S), test_impl(S))
+
+    def test_series_loc_value_int(self):
+        def test_impl(S):
+            return S.loc[2]
+
+        jit_impl = hpat.jit(test_impl)
+
+        S = pd.Series([11, 22, 33], [2, 3, 5])
+        self.assertEqual(jit_impl(S), test_impl(S))
+
+    def test_series_loc_value_str(self):
+        def test_impl(S):
+            return S.loc['2']
+
+        jit_impl = hpat.jit(test_impl)
+
+        S = pd.Series([11, 22, 33], ['2', '3', '5'])
+        self.assertEqual(jit_impl(S), test_impl(S))
+
+    def test_series_loc_slice_int(self):
+        def test_impl(S):
+            return S.loc[2:5]
+
+        jit_impl = hpat.jit(test_impl)
+
+        S = pd.Series([11, 22, 33], [2, 3, 5])
+        self.assertEqual(jit_impl(S), test_impl(S))
+
+    def test_series_loc_slice_str(self):
+        def test_impl(S):
+            return S.loc['2':'5']
```

Comment:
@densmirn Could you please help to combine similar tests into one with loop over diffrent input data?

Reply:
@shssf the similar tests were combined, but the tests are failing as before. So @1e-to could you fix all issues to get all tests passed?